### DISCUSSION TITLE
Additional button to Navigation terminal that sends continuous signal.

### DISCUSSION THREAD

**Discussion by Singularia** [FakeFishGames/Barotrauma#8210] [Ideas and suggestions]
Standart Navigation Terminal has 3 buttons which often help to save space (no need to place buttons and labels) and centralize control into one UI panel. Use-cases for buttons can be (not limited to):

- Sonar controlled cannons
- All-Door/Door close/open
- Multiple drones undock (1 for each button)
- Reactor shutdown
- Battery-Reactor power switch

But there are cases when **some player-created circuits may require a continuous 1 or 0 signal instead of the pulse signal**. In such cases what is available to builders and in-game engineers is either use of levers or switches _(when them and labels)_ or creating a pulse to continuous circuit workaround, both of these variants are not an ideal solution.

I propose to **add an additional button that will have the functionality of an already existing Switch or Lever**, it basically will send continuous 1 or 0 instead of a single pulse.
It can be added as a separate button _(separate from the other 3 pulse buttons)_ or placed directly under them. As for the visual indication of what it sends at the moment, I think **the effect of a "pushed" button** (like when we press and hold buttons of the Nav Terminal) **will work as a good status indicator**. _(1 = "pressed", 0 = "released")_

[_An idea of how it may look like_]
![image](https://user-images.githubusercontent.com/62324831/153009341-1b8aa4c1-dab9-43a6-8c2c-d24770611dc4.png)

In short, even one additional button that sends continuous 1/0 will help to save space and give the ability to control circuits that require `SET_STATE` directly from Navigation Terminal.

**Comment by Singularia** [comment]
In some earlier versions of a game, I saw an additional tick box under buttons. Even though this probably was an empty button that did nothing, the idea itself IMO will be good and useful for most, if not all players to consider implementing.
![image](https://user-images.githubusercontent.com/62324831/153015414-56b2f428-5f12-46c8-8b54-35887c6bbd55.png)
